File: Octagon-Project/Octagon/Form1.vb
Public Class Form1
    Dim angle As Double = 0
    Dim findClassPath As New FindClassPath
    Dim p1 As Image = Image.FromFile(findClassPath.find & "\Resources\spartan1.png")
    Dim p2 As Image = Image.FromFile(findClassPath.find & "\Resources\spartan2.png")
    Dim p3 As Image = Image.FromFile(findClassPath.find & "\Resources\spartan3.png")
    Dim p4 As Image = Image.FromFile(findClassPath.find & "\Resources\spartan4.png")
    Dim p5 As Image = Image.FromFile(findClassPath.find & "\Resources\spartan5.png")
    Dim p6 As Image = Image.FromFile(findClassPath.find & "\Resources\spartan6.png")
    Dim p7 As Image = Image.FromFile(findClassPath.find & "\Resources\spartan7.png")
    Dim p8 As Image = Image.FromFile(findClassPath.find & "\Resources\spartan8.png")
    Dim x As Double
    Dim y As Double
    Dim numx As Double = 0
    Dim numy As Double = 0

    Private Sub Form1_KeyDown(sender As Object, e As KeyEventArgs) Handles Me.KeyDown

        Select Case (e.KeyCode)
            Case Keys.Left
                angle -= 22.5 Mod 360
                x = ((map.Width / 2.5) * Math.Cos(getRadiansFromDegree(angle))) - spartan.Width / 2
                y = ((map.Height / 2.5) * Math.Sin(getRadiansFromDegree(angle))) - spartan.Height / 2
            Case Keys.Right
                angle += 22.5 Mod 360
                x = ((map.Width / 2.5) * Math.Cos(getRadiansFromDegree(angle))) - spartan.Width / 2
                y = ((map.Height / 2.5) * Math.Sin(getRadiansFromDegree(angle))) - spartan.Height / 2
            Case Keys.W
                spartan.Top -= 10
            Case Keys.S
                spartan.Top += 10
            Case Keys.A
                spartan.Left -= 10
            Case Keys.D
                spartan.Left += 10
        End Select

        ' spartan.Image = p1
        'spartan.Location() = New Point(x + Me.Width / 2, y + (Me.Height / 2) - (spartan.Height / 2))

        If angle < 0 Then angle = 360 - angle
        'Select Case (angle Mod 360)
        '    Case 0 To 44 ', -315 To -359
        '        spartan.Image = p1
        '        'spartan.Location = setPoint(2, 1)
        '        'spartan.Top -= spartan.Height
        '        TextBox1.Text = "case 1"
        '    Case 45 To 89 ', -270 To -314
        '        spartan.Image = p2
        '        TextBox1.Text = "case 2"
        '    'spartan.Location = setPoint(3, 1)
        '    Case 90 To 134 ', -225 To -269
        '        spartan.Image = p3
        '        'spartan.Location = setPoint(3, 2)
        '        'spartan.Left += spartan.Width
        '        TextBox1.Text = "case 3"
        '    Case 135 To 179 ', -180 To -224
        '        spartan.Image = p4
        '        'spartan.Location = setPoint(3, 3)
        '        TextBox1.Text = "case 4"
        '    Case 180 To 224 ', -135 To -179
        '        spartan.Image = p5
        '        'spartan.Location = setPoint(2, 3)
        '        'spartan.Top += spartan.Height
        '        TextBox1.Text = "case 5"
        '    Case 225 To 269 ', -90 To -134
        '        spartan.Image = p6
        '        'spartan.Location = setPoint(1, 3)
        '        TextBox1.Text = "case 6"
        '    Case 270 To 314 ', -45 To -89
        '        spartan.Image = p7
        '        'spartan.Location = setPoint(1, 2)
        '        'spartan.Left -= spartan.Width
        '        TextBox1.Text = "case 7"
        '    Case 315 To 359 ', 0 To -44
        '        spartan.Image = p8
        '        'spartan.Location = setPoint(1, 1)
        '        TextBox1.Text = "case 8"

        '        'Case Else
        '        'spartan.Image = p1
        '        'spartan.Location = setPoint(1, 1)
        'End Select

        TextBox2.Text = "angle: " & angle Mod 360
        'spartan.Left += spartan.Width / 7
        'spartan.Top -= spartan.Height / 7
        spartan.Refresh()
    End Sub

    Private Function setPoint(mx As Integer, my As Integer)
        Dim x As Integer = ((Me.Width / 4) * mx) - (spartan.Width / 1.4)
        Dim y As Integer = ((Me.Height / 4) * my) - (spartan.Height / 1.4)
        Return New Point(x, y)
    End Function

    Private Function getRadiansFromDegree(degree As Double) As Double
        Dim radians As Double = (degree * Math.PI) / 180.0
        Return radians
    End Function

    Private Sub Form1_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Me.Focus()
        spartan.Parent = map
        spartan.BackColor = Color.Transparent
        Timer1.Start()
    End Sub

    Private Sub Timer1_Tick(sender As Object, e As EventArgs) Handles Timer1.Tick
        Dim cx As Integer = Cursor.Position.X
        Dim cy As Integer = Cursor.Position.Y
        Dim cc As String = cx & "," & cy
        TextBox3.Text = cc

        If Cursor.Position.Y < (Me.Top + spartan.Top) And Cursor.Position.X > (Me.Left + spartan.Left) And Cursor.Position.X < (Me.Left + spartan.Left + spartan.Width) Then
            spartan.Image = p1
        ElseIf Cursor.Position.Y < (Me.Top + spartan.Top) And Cursor.Position.X > (Me.Left + spartan.Left + spartan.Width) Then
            spartan.Image = p2
        ElseIf Cursor.Position.Y > (Me.Top + spartan.Top) And Cursor.Position.Y < (Me.Top + spartan.Top + spartan.Height) And Cursor.Position.X > (Me.Left + spartan.Left + spartan.Width)
            spartan.Image = p3
        ElseIf Cursor.Position.Y > (Me.Top + spartan.Top) And Cursor.Position.X > (Me.Left + spartan.Left + spartan.Width)
            spartan.Image = p4
        ElseIf Cursor.Position.Y > (Me.Top + spartan.Top) And Cursor.Position.X > (Me.Left + spartan.Left) And Cursor.Position.X < (Me.Left + spartan.Left + spartan.Width)
            spartan.Image = p5
        ElseIf Cursor.Position.Y > (Me.Top + spartan.Top + spartan.Height) And Cursor.Position.X < (Me.Left + spartan.Left)
            spartan.Image = p6
        ElseIf Cursor.Position.Y > (Me.Top + spartan.Top) And Cursor.Position.Y < (Me.Top + spartan.Top + spartan.Height) And Cursor.Position.X < (Me.Left + spartan.Left)
            spartan.Image = p7
        ElseIf Cursor.Position.Y < (Me.Top + spartan.Top) And Cursor.Position.X < (Me.Left + spartan.Left)
            spartan.Image = p8
        End If
 
    End Sub

    Private Sub map_Click(sender As Object, e As EventArgs) Handles map.Click

    End Sub
End Class

File: OctagonC/OctagonC/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using System.Drawing.Drawing2D;

namespace WindowsFormsApplication2{
    
    public partial class Form1 : Form
    {
        private static FindClassPath fcp = new FindClassPath();
        private static Image p1;
        private static Image p2;
        private static Image p3;
        private static Image p4;
        private static Image p5;
        private static Image p6;
        private static Image p7;
        private static Image p8;
        private static Image octa;
        double angle;
        bool W, A, S, D = false;
        private Bitmap spartano;
        private Graphics gfx;
        private int x = 199;
        private int y = 16;
        private int direction = 1;

        public Form1() {
            p1 = Image.FromFile(fcp.find() + "spartan1.png");
            p2 = Image.FromFile(fcp.find() + "spartan2.png");
            p3 = Image.FromFile(fcp.find() + "spartan3.png");
            p4 = Image.FromFile(fcp.find() + "spartan4.png");
            p5 = Image.FromFile(fcp.find() + "spartan5.png");
            p6 = Image.FromFile(fcp.find() + "spartan6.png");
            p7 = Image.FromFile(fcp.find() + "spartan7.png");
            p8 = Image.FromFile(fcp.find() + "spartan8.png");
            octa = Image.FromFile(fcp.find() + "map.png");

            InitializeComponent();
            map.Controls.Add(spartan);
            spartan.BackColor = Color.Transparent;
            this.Focus();
            timer1.Start();
            timer2.Start();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
        }
       
        void ReadKey(object sender, KeyEventArgs e)
        {
            switch ((e.KeyCode))
            {
                case Keys.W:
                    W = true;
                    break;
                case Keys.S:
                    S = true;
                    break;
                case Keys.A:
                    A = true;
                    break;
                case Keys.D:
                    D = true;
                    break;
                    
                    //angle -= 22.5 % 360;
                    //x = ((map.Width / 2.5) * Math.Cos(getRadiansFromDegree(angle-90))) - spartan.Width / 2;
                    //y = ((map.Height / 2.5) * Math.Sin(getRadiansFromDegree(angle-90)));

                    //spartan.Location = new Point(Convert.ToInt16(x + Width / 2), Convert.ToInt16(y + (Height / 2) - (spartan.Height / 2)));
            }

            if (angle < 0)
            {
                angle = 360 - angle;
            }

            TextBox2.Text = "angle: " + angle % 360;
            spartan.Refresh();
        }

        private void timer1_Tick(object sender, EventArgs e)
        {
            int cx = Cursor.Position.X;
            int cy = Cursor.Position.Y;
            string cc = cx + "," + cy;
            TextBox3.Text = cc;
            
            if (Cursor.Position.Y < (this.Top + spartan.Top) 
              & Cursor.Position.X > (this.Left + spartan.Left)                     // Compreso tra la X iniziale e finale dello spartan
              & Cursor.Position.X < (this.Left + spartan.Left + spartan.Width))    // Compreso tra la X iniziale e finale dello spartan
            {
                direction = 1;
                //spartan.Image = p1;
            }

            else if (Cursor.Position.Y < (this.Top + spartan.Top) 
                   & Cursor.Position.X > (this.Left + spartan.Left + spartan.Width))
            {
                direction = 2;
                //spartan.Image = p2;
            }

            else if (Cursor.Position.Y > (this.Top + spartan.Top)                    // Compreso tra la Y iniziale e finale dello spartan
                   & Cursor.Position.Y < (this.Top + spartan.Top + spartan.Height)      // Compreso tra la Y iniziale e finale dello spartan
                   & Cursor.Position.X > (this.Left + spartan.Left + spartan.Width))
            {
                direction = 3;
                //spartan.Image = p3;
            }

            else if (Cursor.Position.Y > (this.Top + spartan.Top) 
                   & Cursor.Position.X > (this.Left + spartan.Left + spartan.Width))
            {
                direction = 4;
                //spartan.Image = p4;
            }

            else if (Cursor.Position.Y > (this.Top + spartan.Top+spartan.Height) 
                   & Cursor.Position.X > (this.Left + spartan.Left)                     // Compreso tra la X iniziale e finale dello spartan
                   & Cursor.Position.X < (this.Left + spartan.Left + spartan.Width))    // Compreso tra la X iniziale e finale dello spartan
            {
                direction = 5;
                //spartan.Image = p5;
            }

            else if (Cursor.Position.Y > (this.Top + spartan.Top + spartan.Height) 
                   & Cursor.Position.X < (this.Left + spartan.Left))
            {
                direction = 6;
                //spartan.Image = p6;
            }

            else if (Cursor.Position.Y > (this.Top + spartan.Top)                    // Compreso tra la Y iniziale e finale dello spartan
                   & Cursor.Position.Y < (this.Top + spartan.Top + spartan.Height)      // Compreso tra la Y iniziale e finale dello spartan
                   & Cursor.Position.X < (this.Left + spartan.Left))
            {
                direction = 7;
                //spartan.Image = p7;
            }

            else if (Cursor.Position.Y < (this.Top + spartan.Top) 
                   & Cursor.Position.X < (this.Left + spartan.Left))
            {
                direction = 8;
                //spartan.Image = p8;
            }

            creaSpartano(new Bitmap(fcp.find() + "spartan" + direction + ".png"), x, y);


        }
        private object setPoint(int mx, int my)
        {
            int x = Convert.ToInt16(((this.Width / 4) * mx) - (spartan.Width / 1.4));
            int y = Convert.ToInt16(((this.Height / 4) * my) - (spartan.Height / 1.4));
            return new Point(x, y);
        }

        private double getRadiansFromDegree(double degree)
        {
            double radians = (degree * Math.PI) / 180;
            return radians;
        }
        void ReadKeyUp(object sender, KeyEventArgs e)
        {
            switch (e.KeyCode)
            {
                case Keys.W:
                    W = false;
                    break;
                case Keys.S:
                    S = false;
                    break;
                case Keys.A:
                    A = false;
                    break;
                case Keys.D:
                    D = false;
                    break;
            }
            TextBox1.Text = Convert.ToString(direction);
            //creaSpartano(new Bitmap(fcp.find() + "spartan" + direction + ".png"), x, y);//spartan.Top -= 10;

        }

        private void timer2_Tick(object sender, EventArgs e)
        {
            if (W) y -= 10; 
            if (A) x -= 10; //spartan.Left -= 10;
            if (S) y += 10; //spartan.Top += 10;
            if (D) x += 10; //spartan.Left += 10;
        }

        private Graphics creaSpartano(Bitmap spartano, int x, int y) {
            gfx = map.CreateGraphics();
            //gfx.Clear(Form1.ActiveForm.BackColor);
            gfx.DrawImage(spartano, new Rectangle(x, y, spartano.Width/8, spartano.Height/8));
            return gfx;
        }
    }
}
